src/main/java/net/sf/jabref/gui/JabRefFrame.java

```diff
@@ -1646,14 +1644,12 @@ public void addTab(BasePanel bp, File file, boolean raisePanel) {
                 title = title + '*';
             }
         } else {
-            title = file.getName();
+            title = file.getName() + " (" + bp.database.getEntryCount() + ")";
```

Comment:
One minor suggestion: the code to determine the title is distributed in two places. One could create a method in BasePanel which computes the title and just reuse the method from there.